tasks/managed/filter-already-released-images/filter-already-released-images.yaml

```diff
@@ -0,0 +1,344 @@
+            
+            if [ "${NUM_REPOS}" -eq 0 ]; then
+                echo "WARNING: No repositories found for component ${COMPONENT_NAME}"
+                echo "  Component will be kept (not filtered) since there are no target repos to check"
+                FILTERED_COMPONENTS=$(jq --argjson comp "${COMPONENT}" \
+                  '. += [$comp]' <<< "${FILTERED_COMPONENTS}")
+                continue
+            fi
+            
+            echo "Checking component: ${COMPONENT_NAME} (${NUM_REPOS} target repositories)"
+
+            # Check if ALL required tags exist with correct digest in ALL target repositories
+            # Start by assuming all repos are complete, then set to false if any repo is incomplete
+            ALL_TAGS_COMPLETE="true"
+            
+            for ((j=0; j<NUM_REPOS; j++)); do
+                REPO_OBJ=$(jq -c ".[$j]" <<< "${REPOSITORIES}")
+                REPO_URL=$(jq -r '.url // ""' <<< "${REPO_OBJ}")
+                REPO_TAGS=$(jq -c '.tags // []' <<< "${REPO_OBJ}")
+                
+                if [ -z "${REPO_URL}" ]; then
+                    echo "  WARNING: Repository #$((j+1)) has empty URL, skipping"
+                    ALL_TAGS_COMPLETE="false"
+                    continue
+                fi
+
+                NUM_TAGS=$(jq 'length' <<< "${REPO_TAGS}")
+                
+                if [ "${NUM_TAGS}" -eq 0 ]; then
+                    echo "  WARNING: Repository ${REPO_URL} has no tags specified, skipping"
+                    ALL_TAGS_COMPLETE="false"
+                    continue
+                fi
+                
+                echo "  Checking repository: ${REPO_URL} (${NUM_TAGS} tags)"
+                
+                REPO_COMPLETE="true"
+                for ((k=0; k<NUM_TAGS; k++)); do
+                    TAG=$(jq -r ".[$k]" <<< "${REPO_TAGS}")
+                    TARGET_IMAGE="${REPO_URL}:${TAG}"
+                    
+                    # Try to create auth file for target registry (optional for public/test registries)
+                    TARGET_AUTH_FILE=$(mktemp)
+                    if ! select-oci-auth "${REPO_URL}" > "${TARGET_AUTH_FILE}" 2>/dev/null || \
+                       [ ! -s "${TARGET_AUTH_FILE}" ]; then
+                        # No auth available, use empty config
+                        echo '{}' > "${TARGET_AUTH_FILE}"
+                    fi
+                    
+                    # Use oras resolve (same as push-snapshot) to get the manifest digest
+                    # This correctly handles multi-arch images
+                    TARGET_ERROR_FILE=$(mktemp)
+                    if ! ACTUAL_DIGEST=$(oras resolve --registry-config "${TARGET_AUTH_FILE}" \
+                        "${TARGET_IMAGE}" 2>"${TARGET_ERROR_FILE}"); then
+                        TARGET_ERROR_MSG=$(cat "${TARGET_ERROR_FILE}")
+                        rm -f "${TARGET_ERROR_FILE}" "${TARGET_AUTH_FILE}"
```

Comment:
Do we need this here? We already rm on line 260 and 281

Reply:
yep, we need it, it's cleaning up different temporary files;

The one you mentioned is dedicated to the cleanup of COMPONENT_AUTH_FILE and COMPONENT_ERROR_FILE. The current is for the TARGET_AUTH_FILE and TARGET_ERROR_FILE